client/services/http-service.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * This Class is a very minimal wrapper around fetch
+ * to make API calls easy.
+ */
+
+import { RequestInfo, RequestInit, Response } from 'node-fetch';
+
+let fetch: (url: RequestInfo, init?: RequestInit) => Promise<Response>;
+
+if (typeof window === 'undefined') {
+  fetch = require('node-fetch');
+} else {
+  fetch = window.fetch ? window.fetch : require('whatwg-fetch');
+}
```

Comment:
I am not able to get any better/easier way to conditionally load `fetch` proxies. If anyone know about some custom webpack configs that we can use to make this code even more simpler, it would be great.

There is [this PR](https://github.com/zeit/next.js/pull/5159), which looks like a sugarcoat over `typeof window === 'undefined'`. So it doesn't seem to solve my problem.

Reply:
Why did we go with this instead of using `isomorphic-fetch` or something else that is plug and play. If I remember correctly it was because it "wasn't maintained", honestly why would you need to maintain `axios` or `isomorphic-fetch`?

This seems like a really hacky solution to me :confused:

Reply:
@Sonicrida shared some stats on vulnerability issues found in `axios`, which I recently found that was a false alarm from `snyk` tests ([Details here](https://github.com/axios/axios/issues/2447)). I am ok with using anyone of these, but there has been conflicts between other members. I am thinking to open this as an issue and resolve it for all (removing this from blocker).

Reply:
Discussions also concluded that, non maintained libs have many essential issues opened for a long time and haven't been looked into, where `isomorphic-fetch` hasn't been maintained for last 4 years.

Reply:
https://github.com/lquixada/cross-fetch is a well maintained alternative

Reply:
Resolving this as per decisions made - https://github.com/freeCodeCamp/chapter/issues/258